src/libs/Console/index.ts

```diff
@@ -0,0 +1,104 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import {addLog} from '@libs/actions/Console';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {Log} from '@src/types/onyx';
+
+/* store the original console.log function so we can call it */
+// eslint-disable-next-line no-console
+const originalConsoleLog = console.log;
+
+/* List of patterns to ignore in logs. "logs" key always needs to be ignored because otherwise it will cause infinite loop */
+const logPatternsToIgnore = [`merge() called for key: ${ONYXKEYS.LOGS}`];
+
+/**
+ * Check if the log should be attached to the console
+ * @param message the message to check
+ * @returns true if the log should be attached to the console
+ */
+function shouldAttachLog(message: string) {
+    return !logPatternsToIgnore.some((pattern) => message.includes(pattern));
+}
+
+/**
+ * Goes through all the arguments passed the console, parses them to a string and adds them to the logs
+ * @param args the arguments to log
+ */
+function logMessage(args: unknown[]) {
+    const message = args
+        .map((arg) => {
+            if (typeof arg === 'object') {
+                try {
+                    return JSON.stringify(arg, null, 2); // Indent for better readability
+                } catch (e) {
+                    return 'Unserializable Object';
+                }
+            }
+
+            return String(arg);
+        })
+        .join(' ');
+    const newLog = {time: new Date(), level: CONST.DEBUG_CONSOLE.LEVELS.INFO, message};
+    addLog(newLog);
+}
+
+/**
+ * Override the console.log function to add logs to the store
+ * @param args arguments passed to the console.log function
+ */
+// eslint-disable-next-line no-console
+console.log = (...args) => {
+    logMessage(args);
+    originalConsoleLog.apply(console, args);
+};
+
+const charsToSanitize = /[\u2018\u2019\u201C\u201D\u201E\u2026]/g;
+
+const charMap: Record<string, string> = {
```

Comment:
Maybe move this to before `charsToSanitize` variable so we can map out unicode to string literals.